src/object.js

```diff
@@ -146,11 +161,40 @@ ReflectionObject.prototype.onRemove = function onRemove(parent) {
 ReflectionObject.prototype.resolve = function resolve() {
     if (this.resolved)
         return this;
+    if (this instanceof Root || this.parent && this.parent.resolved)
+        this._resolveFeatures();
     if (this.root instanceof Root)
-        this.resolved = true; // only if part of a root
+        this.resolved = true;
     return this;
 };
 
+/**
+ * Resolves child features from parent features
+ * @returns {undefined}
+ */
+ReflectionObject.prototype._resolveFeatures = function _resolveFeatures() {
+    var defaults = {};
+
+    if (this instanceof Root) {
+        if (this.root.getOption("syntax") === "proto2") {
+            defaults = Object.assign({}, proto2Defaults);
+        } else if (this.root.getOption("syntax") === "proto3") {
+            defaults = Object.assign({}, proto3Defaults);
+        } else if (this.root.getOption("edition") === "2023") {
+            defaults = Object.assign({}, editions2023Defaults);
+        }
```

Comment:
We should probably throw for unknown editions to clarify they aren't supported yet

Reply:
We throw in `parseEdition` parse.js line 289